RawSpeed/DngDecoderSlices.cpp

```diff
@@ -143,6 +143,159 @@ my_error_throw (j_common_ptr cinfo)
   ThrowRDE("JPEG decoder error!");
 } 
 
+static void decodeFPDeltaRow(unsigned char * src, unsigned char * dst, size_t tileWidth, size_t realTileWidth, unsigned int bytesps, int factor) {
+  // DecodeDeltaBytes
+  for (size_t col = factor; col < realTileWidth*bytesps; ++col) {
+    src[col] += src[col - factor];
+  }
+  // Reorder bytes into the image
+  // 16 and 32-bit versions depend on local architecture, 24-bit does not
+  if (bytesps == 3) {
+    for (size_t col = 0; col < tileWidth; ++col) {
+      dst[col*3] = src[col];
+      dst[col*3 + 1] = src[col + realTileWidth];
+      dst[col*3 + 2] = src[col + realTileWidth*2];
+    }
+  } else {
+    union X { unsigned int x; unsigned char c; };
+    if (((union X){1}).c) {
+    for (size_t col = 0; col < tileWidth; ++col) {
+      for (size_t byte = 0; byte < bytesps; ++byte)
+        dst[col*bytesps + byte] = src[col + realTileWidth*(bytesps-byte-1)];  // Little endian
+    }
+    } else {
+    for (size_t col = 0; col < tileWidth; ++col) {
+      for (size_t byte = 0; byte < bytesps; ++byte)
+        dst[col*bytesps + byte] = src[col + realTileWidth*byte];
+        }
+      }
+    }
+}
+
+// From DNG SDK dng_utils.h
```

Comment:
I'd personally try really hard to find a better-licensed version of this function.
After all, it looks like it simply converts half-float to full float.
Probably OpenEXR has such a function.